includes/class-newspack-ads-settings.php

```diff
@@ -215,18 +221,40 @@ public static function get_settings_list() {
 			),
 		);
 
+		$default_setting = array(
+			'section' => '',
+			'type'    => 'string',
+			'public'  => false,
+		);
+
 		$settings_list = apply_filters( 'newspack_ads_settings_list', $settings_list );
 
+		// Add default settings and get values.
 		$settings_list = array_map(
-			function ( $item ) {
-				$default       = isset( $item['default'] ) ? $item['default'] : false;
-				$item['value'] = get_option( self::get_setting_option_name( $item ), $default );
-				settype( $item['value'], $item['type'] );
+			function ( $item ) use ( $default_setting ) {
+				$item          = array_merge( $default_setting, $item );
```

Comment:
Non-blocking: `wp_parse_args`? It won't do anything different in this case, but I tend to stick with that for merging options arrays with defaults.

Reply:
Good call! Done in a2307b1c6ce35cfa3a023522370a6f606217666c